lib/components/primitive-components/SmtPad.ts

```diff
@@ -26,6 +27,35 @@ export class SmtPad extends PrimitiveComponent<typeof smtPadProps> {
     }
   }
 
+  createSolderPasteComponent(): void {
+    let solderPaste: SolderPaste
+    if (this._parsedProps.shape === "circle") {
+      solderPaste = new SolderPaste({
+        shape: "circle",
+        radius: this._parsedProps.radius * 0.7,
+        layer: this._parsedProps.layer,
+        pcbX: this._parsedProps.pcbX,
+        pcbY: this._parsedProps.pcbY,
+      })
+      solderPaste.pcb_smtpad_id = this.pcb_smtpad_id
+      this.add(solderPaste)
+      solderPaste.doInitialPcbPrimitiveRender()
+    }
+    if (this._parsedProps.shape === "rect") {
+      solderPaste = new SolderPaste({
+        shape: "rect",
+        height: this._parsedProps.height * 0.7,
+        width: this._parsedProps.height * 0.7,
+        layer: this._parsedProps.layer,
+        pcbX: this._parsedProps.pcbX,
+        pcbY: this._parsedProps.pcbY,
+      })
+      solderPaste.pcb_smtpad_id = this.pcb_smtpad_id
+      this.add(solderPaste)
+      solderPaste.doInitialPcbPrimitiveRender()
```

Comment:
i think for simplicity, we shouldn't have pcb primitives contain other pcb primitives, so what we can do here is not use the `SolderPaste` class at all in this class, we'll just insert the `pcb_solder_paste` json inside of our rendering function.

You could remove the `SolderPaste` class completely in the short term for simplicity, it's just part of smtpad rendering (when someone needs it we can introduce the class)